### ISSUE TITLE
Broker Stops working with the Too many files open exception.

### ISSUE THREAD

**Issue by kunalkrishna1111** [zeebe-io/zeebe#3091]
**Describe the bug**
When you run the broker for long, many a times it stops working with an exception.

**To Reproduce**
Run a load test for 4-6 hours in a cluster of 3 brokers and 1 gateway

If possible add a minimal reproducer code sample
 - when using the Java client: https://github.com/zeebe-io/zeebe-test-template-java

**Expected behavior**
Broker should run forever. 

**Log/Stacktrace**
If possible add the full stacktrace or Zeebe log which contains the issue. 

<details><summary>Full Stacktrace</summary>
 <p>

```
03:18:13.759 [] [netty-messaging-event-epoll-server-0] WARN  io.netty.channel.DefaultChannelPipeline - An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
io.netty.channel.unix.Errors$NativeIoException: accept(..) failed: Too many open files
03:18:14.700 [] [nioEventLoopGroup-2-1] WARN  io.netty.channel.DefaultChannelPipeline - An exceptionCaught() event was fired, and it reached at the tail of the pipeline. It usually means the last handler in the pipeline did not handle the exception.
java.io.IOException: Too many open files
	at sun.nio.ch.ServerSocketChannelImpl.accept0(Native Method) ~[?:?]
	at sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:422) ~[?:?]
	at sun.nio.ch.ServerSocketChannelImpl.accept(ServerSocketChannelImpl.java:250) ~[?:?]
	at io.netty.util.internal.SocketUtils$5.run(SocketUtils.java:110) ~[netty-common-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.util.internal.SocketUtils$5.run(SocketUtils.java:107) ~[netty-common-4.1.37.Final.jar:4.1.37.Final]
	at java.security.AccessController.doPrivileged(Native Method) ~[?:?]
	at io.netty.util.internal.SocketUtils.accept(SocketUtils.java:107) ~[netty-common-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.channel.socket.nio.NioServerSocketChannel.doReadMessages(NioServerSocketChannel.java:145) ~[netty-transport-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.channel.nio.AbstractNioMessageChannel$NioMessageUnsafe.read(AbstractNioMessageChannel.java:75) [netty-transport-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:697) [netty-transport-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:632) [netty-transport-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:549) [netty-transport-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:511) [netty-transport-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:918) [netty-common-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.37.Final.jar:4.1.37.Final]
	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.37.Final.jar:4.1.37.Final]
	at java.lang.Thread.run(Thread.java:748) [?:?]
```

</p>
</details>

**Environment:**
 - OS: [e.g. Linux]
 - Zeebe Version: [e.g. 0.20.0]
 - Configuration: [e.g. exporters etc.]

Refer - https://github.com/camundacon2019/zeebe-ft-demo

**Comment by menski** [comment]
@kunalkrishna1111 could you please describe your environment? Which Zeebe version, which OS version and what is your open file limit, i.e. `ulimit -n`?

**Comment by Zelldon** [comment]
This should be fixed with the latest release (0.21), we had a bug where the compaction does not work probably and that why it could happen that many files have been open.
